Automattic/wp-calypso
# Stepper: Remove renderStep dependecy from step router

Depends on #91241

## Proposed Changes
Nowadays we have the FlowRender building a function to manage if step content should or not be rendered.  This function is passed via props to the StepRoute. 

## Why are these changes being made?
The current strategy, it causes coupling between the step route and the flow-render and possible necessary rerenders because the renderStep reference is re-created on every component render, causing unnecessary step-router rerendering.

## Testing Instructions
* Open any step and check if the content is rendered. (the StepRouter's automated tests are checking multiple scenarios around it)

## Pre-merge Checklist
